src/main/java/org/kairosdb/core/opentracing/HttpHeadersCarrier.java

```diff
@@ -0,0 +1,33 @@
+package org.kairosdb.core.opentracing;
+
+import io.opentracing.contrib.web.servlet.filter.HttpServletRequestExtractAdapter;
+import io.opentracing.propagation.TextMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class HttpHeadersCarrier implements TextMap{
+
+    public HttpHeadersCarrier()  {
+        this.httpHeaders = null;
```

Comment:
it should be 
```
this.httpHeaders = new MultivaluedMapImpl();
```
Otherwise you'll get `NullPointerException` when calling `iterator()` after using default constructor.